SingularityBase/src/main/java/com/hubspot/mesos/protos/MesosRangesObject.java

```diff
@@ -0,0 +1,57 @@
+package com.hubspot.mesos.protos;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Optional;
+
+public class MesosRangesObject {
+  private final Optional<List<MesosRangeObject>> range;
+
+  @JsonCreator
+  public MesosRangesObject(@JsonProperty("range") Optional<List<MesosRangeObject>> range) {
+    this.range = range;
+  }
+
+  public List<MesosRangeObject> getRange() {
+    return range.or(Collections.emptyList());
+  }
+
+  @JsonIgnore // to mimic mesos
+  public List<MesosRangeObject> getRangesList() {
+    return getRange();
```

Comment:
not sure I understand the reason for this method. doesn't seem to do anything different than `getRange()`

Reply:
It's to mimic the convention used by the Mesos API, with stuff like [`getPortsList()`](http://mesos.apache.org/api/latest/java/org/apache/mesos/Protos.PortsOrBuilder.html#getPortsList--).

Reply:
correct, added with the aim of not breaking binary compatibility for those who were using the old mesos protos objects